kvs.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+type KVS interface {
+	// Store stores a key value pair.
+	Store(k, v string) error
+
+	// Load returns the value associated with a given key.
+	Load(k string) (string, error)
+}
+
+type MyKVS map[string]string
+
+func NewMyKVS() MyKVS {
+	return MyKVS{}
+}
+
+func (kvs MyKVS) Store(k, v string) error {
+	_, ok := kvs[k]
+	if ok {
+		return MyKVSError{"already exist"}
```

Comment:
You implemented your own error type, it's not wrong but generally we do this for what we call sentinel errors. A sentinel error is a type implementing the `error` interface, one can test against in order to know if the called function returns an error of a specific type. Example of sentinel errors are: `io.EOF` (end of file for streams/files) or context cancelled (https://pkg.go.dev/context#pkg-variables).

Another use case for implementing your own errors is when the error type carries some additional infos as fields of the struct implementing the error type.
Since we are in neither of those cases, we can here simply use:

[fmt.Errorf](https://pkg.go.dev/fmt#Errorf)